internal/services/loganalytics/custompollers/log_analytics_data_export_poller.go

```diff
@@ -0,0 +1,47 @@
+package custompollers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/operationalinsights/2020-08-01/dataexport"
+	"github.com/hashicorp/go-azure-sdk/sdk/client/pollers"
+)
+
+var _ pollers.PollerType = &LogAnalyticsDataExportPoller{}
+
+type LogAnalyticsDataExportPoller struct {
+	client *dataexport.DataExportClient
+	id     dataexport.DataExportId
+}
+
+var (
+	pollingSuccess = pollers.PollResult{
+		Status:       pollers.PollingStatusSucceeded,
+		PollInterval: 10 * time.Second,
+	}
+	pollingInProgress = pollers.PollResult{
+		Status:       pollers.PollingStatusInProgress,
+		PollInterval: 10 * time.Second,
+	}
+)
+
+func NewLogAnalyticsDataExportPoller(client *dataexport.DataExportClient, id dataexport.DataExportId) *LogAnalyticsDataExportPoller {
+	return &LogAnalyticsDataExportPoller{
+		client: client,
+		id:     id,
+	}
+}
+
+func (p LogAnalyticsDataExportPoller) Poll(ctx context.Context) (*pollers.PollResult, error) {
+	resp, err := p.client.Get(ctx, p.id)
+	if err != nil {
```

Comment:
is there a provisioning state returned that we could use here?

Reply:
There is no this kind of property, actually, the creation is a sync action, however the service side has some other process after the process, which might result in 404 response, so added this workaround